endpoints/openrtb2/sample-requests/invalid-native/assets-with-partial-ids.json

```diff
@@ -0,0 +1,17 @@
+{
+    "context": 1,
+    "plcmttype": 1,
+    "assets": [
+        {
+            "id": 1,
+            "img": {
```

Comment:
I am wondering.. is this test case actually for an _invalid_ native request? Based on our assumption that there's gonna be asset IDs either specified or not specified for _all_ assets and that we don't expect partial IDs to be specified, this would actually be a _valid-native_ test case, no? 

I guess it gets a little tricky when we consider assets with partial IDs set. We will treat them as if all asset IDs are set and so the assets that don't have asset IDs set will have zero values and we will therefore consider them not being unique asset IDs and return an error which is a fair behavior. But, if only one of the assets doesn't have an ID set, like in this case, then we won't return an error because one asset with a `0` asset ID is still valid. Therefore, this example shouldn't really return an error, correct?

Reply:
And looks like that is actually one of the tests that's failing:

```
--- FAIL: TestBadNativeRequests (0.38s)
    auction_test.go:175: Expected a 400 response from sample-requests/invalid-native/assets-with-partial-ids.json. Got 200: {"id":"req-id","bidid":"test bid id","nbr":0}
```